Frontend_Web_Development_React_Angular_Vue/Vue/Week 2/Data_Manipulation_using_Methods.md

```diff
@@ -0,0 +1,157 @@
+        "mood": "sad",
+        "fish": "goldfish",
+        "colors": ["green", "red"]
+    }
+]
+```
+
+and from this, you want all the colors from the color array so for this you would play around or manipulate the received data from the specific endpoint. You might use the following piece of code to do so:
+
+```js
+let response = [
+    {
+        mood: "happy",
+        fish: "robin",
+        colors: ["blue", "green"]
+    },
+    {
+        mood: "tired",
+        fish: "panther",
+        colors: ["green", "black", "orange", "blue"]
+    },
+    {
+        mood: "sad",
+        fish: "goldfish",
+        colors: ["green", "red"]
+    }
+]
+let myColors = [];
+for(let i=0; i<response.length; i++) {
+    myColors.push(response[i].colors)
+}
+```
+
+By the above code, you would get all the colors in the _myColors_ variable and you can then manipulate and use it accordingly.
+
+So here the need for data manipulation arises, you just don't want raw data coming from an endpoint or data which is passed as props in some of the JS frameworks, you would want to use that data and display it in the format as the client wants to see it.
+
+### Using methods for data manipulation
+
+In Vue.js to declare custom methods or to write a logic that should run at a particular instance, we write them in the methods object provided by the VUe.js instance. The **this** value is automatically bound to **methods** by Vue.js such that each function in the method has its unique copy of the component instance. Here **this** refers to the component instance.
+
+With the help of **this** keyword, we can access the data properties in the component and manipulate them when any event listener is fired and that particular event matches with that in the methods object.
+
+Just like any other property of Vue component methods are accessible within the component's template. The most common way of using methods in a template is as follows:
+
+```html
+<button @click="changebkg">Click To change background</button>
+```
+
+### Implementation of Data Manipulation using methods
+
+#### Illustration 1
+
+```html
+<template>
+  <div id="app">
+    <span>{{message}}</span>
+    <hr>
+    <button @click="changeMessage()">Click Me</button>
+  </div>
+</template>
+
+
+<script>
+export default {
+  name: "App",
+  data: () => ({
+        message: "Implementation of Data Manipulation"
+  }),
+  methods: {
+    changeMessage () {
+      this.message = "Succesfully message changed"
+    }
+  }
+};
+</script>
+````
+
+In the above code snippet you can see that in the method properties there is a function named **changeMessage()** using this we can manipulate the **message** data property and set it to the text as we want that should be displayed on the screen on clicking the button.
+
+Below is the output of the above performed operation:
+
+<div style="display: flex; justify-content: center; align-items: center">
+  <img src="./assets/data_manipulation_message.gif" style="width: 500px;">
```

Comment:
Your GIF looks like you are clicking twice on the button, because your GIF start from the position of your mouse on button.
Follow the below for GIF :-
Just make GIF in which you are going over the button then clicking on it and then hovering out of the button.